### PR TITLE
Parse VMSS CSE in agentbaker

### PR DESCRIPTION
**Pull request by zjpjack**
Move parse vmss cse logic into agent baker

### CODE REVIEW

File: pkg/agent/const.go [hunk 1]
```diff
@@ -202,3 +202,10 @@ const (
 	// AppGwIngressAddonName appgw addon
 	AppGwIngressAddonName = "appgw-ingress"
 )
+
+const (
+	// VMSSInstanceErrorCode is used to parse vmss instance error code
+	VMSSInstanceErrorCode = "vmssInstanceErrorCode="
```
Comment (bowang-666) [inline pkg/agent/const.go:208]: I think the = sign should be in the message itself

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -704,3 +705,25 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
...
+	var cseStatus datamodel.VMSSInstanceViewCSEStatus
+	var rerr error
+	start := strings.Index(message, "[stdout]") + len("[stdout]")
+	end := strings.Index(message, "[stderr]")
+	if end > start {
+		rawInstanceViewInfo := message[start:end]
+		err := json.Unmarshal([]byte(rawInstanceViewInfo), &cseStatus)
+		if err != nil || cseStatus.ExitCode == "" {
+			// RP needs regex "vmssInstanceErrorCode=" to parse the error
+			rerr = fmt.Errorf("vmssCSE has invalid message=%s, %s%s", message, VMSSInstanceErrorCode, InvalidCSEMessage)
```
Comment (bowang-666) [inline pkg/agent/baker.go:720]: maybe %s=%s?

**Comment by bowang-666** [comment]
lgtm

File: pkg/agent/baker.go [hunk 3]
```diff
@@ -704,3 +705,25 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
...
+func ParseVMSSCSEMessage(message string) (datamodel.VMSSInstanceViewCSEStatus, error) {
+	var cseStatus datamodel.VMSSInstanceViewCSEStatus
+	var rerr error
+	start := strings.Index(message, "[stdout]") + len("[stdout]")
+	end := strings.Index(message, "[stderr]")
+	if end > start {
+		rawInstanceViewInfo := message[start:end]
+		err := json.Unmarshal([]byte(rawInstanceViewInfo), &cseStatus)
+		if err != nil || cseStatus.ExitCode == "" {
+			// RP needs regex "vmssInstanceErrorCode=" to parse the error, please inform RP if you would like to change the regex.
```
Comment (wenwu449) [inline pkg/agent/baker.go:719]: I suggest we do not assume there is only one user of this library, can we say  the error text is part of contract, and client is relying on it?

**Comment by wenwu449** [comment]
please update PR title to be more descriptive

File: pkg/agent/baker.go (hunk 3 above)
Comment (zjpjack) [reply pkg/agent/baker.go]: fix it

**Review by bowang-666 — APPROVED** [review]

**Pull request merged**